components/layout_2020/flow/inline.rs

```diff
@@ -235,48 +284,21 @@ impl<'box_tree, 'a, 'b> InlineFormattingContextState<'box_tree, 'a, 'b> {
                 TextAlign::Start
             },
         };
-        let move_by = match text_align {
-            TextAlign::Start => Length::zero(),
-            TextAlign::Center => (containing_block.inline_size - state.inline_position) / 2.,
-            TextAlign::End => containing_block.inline_size - state.inline_position,
-        };
-        if move_by > Length::zero() {
-            for fragment in &mut fragments {
-                fragment.offset_inline(&move_by);
-            }
-        }
-
-        let size = Vec2 {
-            inline: containing_block.inline_size,
-            block: state.max_block_size,
-        };
-
-        let start_corner = self.next_line_start_position.clone();
-        self.next_line_start_position = Vec2 {
-            inline: Length::zero(),
-            block: self.next_line_start_position.block + size.block,
-        };
-
-        if let Some(sequential_layout_state) = sequential_layout_state {
-            sequential_layout_state.advance_block_position(size.block);
-        }
 
-        let line_had_content =
-            !fragments.is_empty() || state.positioning_context.len() != positioning_context_length;
-        if line_had_content {
-            state
-                .positioning_context
-                .adjust_static_position_of_hoisted_fragments_with_offset(
-                    &start_corner,
-                    positioning_context_length,
-                );
-
-            self.fragments
-                .push(Fragment::Anonymous(AnonymousFragment::new(
-                    Rect { start_corner, size },
-                    fragments,
-                    containing_block.style.writing_mode,
-                )));
+        let line_start_position_from_indent = self.current_line_start_position.inline;
+        let real_line_length_without_indent = self.current_inline_position - whitespace_trimmed;
+        match text_align {
+            TextAlign::Start => line_start_position_from_indent,
+            TextAlign::End => {
+                containing_block.inline_size - real_line_length_without_indent +
```

Comment:
Is the calculation here correct?
For [this test html](http://software.hixie.ch/utilities/js/live-dom-viewer/saved/11941), both master and legacy-layout have the same output as Firefox and Chrome, but this PR doesn't match those.

Reply:
The calculation was incorrect! I have updated this in a way that I think corrects the case you mentioned and also have put a specification link and quote explaining why it's correct now.

Reply:
The `text-align: end` case  doesn't seem to work similar to the other layouts & master for [this case](http://software.hixie.ch/utilities/js/live-dom-viewer/?saved=11943).

IIUC, the spec says 'text-indent' is treated as margin at the 'start'  edge of the linebox and 'text-align: end' will align the contents to the 'end' edge of the box, so the margin cannot not influence the end edge unless the margin exceeds the available width.